### PR TITLE
feat: hesai ptp setup parameters

### PR DESCRIPTION
**Pull request by amc-nu**
## PR Type

- New Feature

## Description

Add support to the hesai hw interface to setup the Lidar's PTP profile.

## Review Procedure

You can now set up the PTP profile using the following arguments in the XML and Python launch files.

```
<arg name="ptp_profile" default="1588v2" description="1588v2|802.1as|automotive"/>
<arg name="ptp_domain" default="0" description="PTP Domain [0-127]."/>
<arg name="ptp_transport_type" default="UDP" description="1588v2 supports 'UDP' or 'L2', other profiles only L2 (HW)"/>
```

## Remarks

- Make sure you have a PTP server running on the host PC
- Only some of the sensors support all the available profiles. Please verify the respective manuals to the corresponding model you are testing with.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

**Comment by drwnz** [comment]
I tested PTP operation and the xml launchfiles. The Python launchfile does not appear to work. It does not set the rotation speed, which has a default as a non-integer causing a launch error. I will submit a PR for the rotation error issue, but the python launcher should probably access more setting (like rotation speed, destination IP).

**Review by drwnz — CHANGES_REQUESTED** [review]

File: nebula_common/include/nebula_common/hesai/hesai_common.hpp [hunk 1]
```diff
@@ -72,28 +66,32 @@ struct HesaiCalibrationConfiguration : CalibrationConfigurationBase
...
+        continue;
+      }
+
+      try {
+        int laser_id = std::stoi(actual_tokens[0]);
+        float elevation = std::stof(actual_tokens[1]);
+        float azimuth = std::stof(actual_tokens[2]);
+        elev_angle_map[laser_id - 1] = elevation;
+        azimuth_offset_map[laser_id - 1] = azimuth;
+        std::cout << "laser " << laser_id << ", elevation " << elevation << ", azimuth " << azimuth << std::endl;
```
[the remark this answers is not in the document]
Comment (amc-nu) [reply nebula_common/include/nebula_common/hesai/hesai_common.hpp]: Indeed, this is a ROS-agnostic module. I removed the verbosity in: https://github.com/tier4/nebula/pull/110/commits/1d59276315b1274bf0b6cbdfe568098cbdc1dcc4

File: nebula_hw_interfaces/src/nebula_hesai_hw_interfaces/hesai_hw_interface.cpp [hunk 2]
```diff
@@ -2993,30 +3000,46 @@ HesaiStatus HesaiHwInterface::CheckAndSetConfig(
         SetSyncAngle(sync_flg, scan_phase);
       });
       t.join();
+      std::this_thread::sleep_for(100ms);
```
[the remark this answers is not in the document]
Comment (amc-nu) [reply nebula_hw_interfaces/src/nebula_hesai_hw_interfaces/hesai_hw_interface.cpp]: Added in https://github.com/tier4/nebula/pull/110/commits/fe8d2af31ffde899ba6e957aeb507c30372b9d80

File: nebula_ros/launch/velodyne_launch_all_hw.xml [hunk 3]
```diff
@@ -28,40 +28,40 @@
...
+<!--        <param name="frame_id" value="$(var frame_id)"/>-->
+<!--        <param name="host_ip" value="$(var host_ip)"/>-->
+<!--        <param name="data_port" value="$(var data_port)"/>-->
+<!--        <param name="gnss_port" value="$(var gnss_port)"/>-->
+<!--        <param name="packet_mtu_size" value="$(var packet_mtu_size)"/>-->
+<!--        <param name="rotation_speed" value="$(var rotation_speed)"/>-->
+<!--        <param name="cloud_min_angle" value="$(var cloud_min_angle)"/>-->
+<!--        <param name="cloud_max_angle" value="$(var cloud_max_angle)"/>-->
+<!--        <param name="setup_sensor" value="$(var setup_sensor)"/>-->
+<!--    </node>-->
```
[the remark this answers is not in the document]
Comment (amc-nu) [reply nebula_ros/launch/velodyne_launch_all_hw.xml]: fixed in https://github.com/tier4/nebula/pull/110/commits/e3175a8908012ac8e75d4a13883edb85c88a3098

**Review by drwnz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**